mdsuite/calculators/green_kubo_distinct_diffusion_coefficients.py

```diff
@@ -170,6 +171,112 @@ def __call__(
 
         self.combinations = list(itertools.combinations_with_replacement(self.species, 2))
 
+    def _correlate(self, ds_a: np.ndarray, ds_b: np.ndarray) -> np.ndarray:
```

Comment:
These functions do not use ``self`` and seem very useful for other applications as well. For the future it would be a good idea to have them as free functions available to all calculators

Reply:
I fully agree. I was thinking of a Green-Kubo parent class or even working out the past templating for these methods. it is the fist time I have put them into MDSuite but they are also capable of being mapped in parallel over devices, e.g. GPUs. So when we look into mulit-gpu or multi-node analysis this will be a nice thing to have

Reply:
That is 3 already steps ahead of what I was proposing, which was just to move the function as it is into sth like a ``math.py`` oder ``calculator_utils.py`` file

Reply:
I think beforehand I would like to think about how that wil look at the best way to handle it as it is similar in the Einstein case. I will try to tackle it tomorrow.

Reply:
I have moved them to the calculator helper methods directory and added standalone tests